packages/web3-utils/src/big_decimal.ts

```diff
@@ -0,0 +1,51 @@
+export class BigDecimal {
+	private readonly bigint: bigint;
+	private readonly decimals: number;
+
+	public constructor(value: bigint | string, decimals?: number) {
+		if (typeof value === 'string' && decimals) {
+			throw new Error('Can not specify decimal with string value');
```

Comment:
If we're padding the number with zeros here:

```typescript
this.bigint = BigInt(`${integer}${fraction.padEnd(this.decimals, '0')}`);
```

why can't we support strings and decimals? e.g.:

```typescript
new BigDecimal('4.2', 5) // '420000'
```

Reply:
If we choose to support this, the padding logic:

```typescript
this.bigint = BigInt(`${integer}${fraction.padEnd(this.decimals, '0')}`);
```

needs to be updated to account for the number of decimals already present. So it should know that `'4.2'` already has a one decimal and the desired amount is `5`, so add `4` zeros. Currently it would add `5`

Reply:
If we support `string` with `decimals` it will be confusing.

Reply:
I think it would be a nice to have for users, but since this isn't going to be exposed publicly, than it's fine not to support it